src/include/common/types/types.h

```diff
@@ -445,21 +446,22 @@ using logical_type_vec_t = std::vector<LogicalType>;
 
 struct VarListType {
     static inline LogicalType* getChildType(const LogicalType* type) {
-        KU_ASSERT(type->getPhysicalType() == PhysicalTypeID::VAR_LIST);
+        KU_ASSERT(type->getPhysicalType() == PhysicalTypeID::VAR_LIST ||
+                  type->getPhysicalType() == PhysicalTypeID::ARRAY);
```

Comment:
Why do we allow `VarListType::getChildType` to be used for `ARRAY`? `ArrayType:: getChildType` should instead be used right? unless I miss something here.

Reply:
Yes, we should be able to scan array from duckdb now.

Reply:
By now, we are using ```VarListColumn``` for ```ARRAY```, and there are many parts of it using ```VarListType``` to get the child type, for example: https://github.com/kuzudb/kuzu/blob/master/src/storage/store/var_list_column.cpp#L62
So I added this as a temporary solution to avoid fixing too much the code of ```VarListColumn```. We should remove this when we finish implementing ```ArrayColumn```.